docs/src/test-fixtures-js.md

```diff
@@ -122,6 +122,11 @@ Fixtures have a number of advantages over before/after hooks:
 - Fixtures are **flexible**. Tests can use any combinations of the fixtures to tailor precise environment they need, without affecting other tests.
 - Fixtures simplify **grouping**. You no longer need to wrap tests in `describe`s that set up environment, and are free to group your tests by their meaning instead.
 
+Three good signs that something is better as a fixture than as a helper function or before/after hook are:
```

Comment:
Looking at the list of features above, and the new items in this PR:
- item 1 echoes the "encapsulate" feature
- item 2 echoes the "composable" feature
- item 3 is close to "on-demand", but is not exactly the same

I wonder whether we should expand the explanation above instead of repeating them here?

Reply:
What I like about your rules of thumb is that they explain fixtures by comparison to concepts familiar to the reader. That really helps building a mental model of when to use fixtures.

So i'm thinking how to weave them into the above while keeping the comparison, but everything I come up with is hard to read. Could you give an example for one of them?

Reply:
Maybe something like this?

```md
- Fixtures **encapsulate** setup and teardown in the same place so it is easier to write. So if you have a before hook setting things up, and an after hook tearing it down, consider turning them into a single fixture.
```

Reply:
That works. What do you think about the wording in https://github.com/microsoft/playwright/pull/33472/commits/117df25856a1107bdfcefd6f6c9a35b00bb75a76?